mithril-common/src/signable_builder/cardano_transactions.rs

```diff
@@ -39,25 +56,51 @@ impl CardanoTransactionsSignableBuilder {
     pub fn new(
         transaction_parser: Arc<dyn TransactionParser>,
         transaction_store: Arc<dyn TransactionStore>,
+        transaction_retriever: Arc<dyn TransactionRetriever>,
         dirpath: &Path,
         logger: Logger,
     ) -> Self {
         Self {
             transaction_parser,
             transaction_store,
+            transaction_retriever,
             logger,
             dirpath: dirpath.to_owned(),
         }
     }
 
     fn compute_merkle_root(&self, transactions: &[CardanoTransaction]) -> StdResult<MKTreeNode> {
-        let store = MKTreeStore::default();
-        let leaves = transactions.iter().map(|tx| tx.into()).collect::<Vec<_>>();
-        let mk_tree = MKTree::new(&leaves, &store)
-            .with_context(|| "CardanoTransactionsSignableBuilder failed to compute MKTree")?;
-        let mk_root = mk_tree
-            .compute_root()
-            .with_context(|| "CardanoTransactionsSignableBuilder failed to compute MKTree root")?;
+        let mut transactions_by_block_ranges: HashMap<BlockRange, Vec<TransactionHash>> =
```

Comment:
Is it make sense to re-implement this method in aggregator ? Is it the same thing that we compute ?

Reply:
The computation will be different when we sign and compute proof, so I'd rather keep code separated first.
However, I have duplicated the `compute_merkle_map_from_transactions` function from the Prover service and added a comment in that sense